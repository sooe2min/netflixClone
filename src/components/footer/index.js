import React from 'react'
import { Container, Title, Row, Column, Link, Select, Text, Copy, Break } from './styles/footer'

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Footer.Title = function FooterTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Footer.Row = function FooterRow({ children, ...restProps }) {
  return <Row {...restProps}>{children}</Row>
}

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>
}

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Footer.Select = function FooterSelect({ children, ...restProps }) {
  return <Select {...restProps}>{children}</Select>
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Footer.Copy = function FooterCopy({ children, ...restProps }) {
  return <Copy {...restProps}>{children}</Copy>
}

Footer.Break = function FooterBreak({ children, ...restProps }) {
  return <Break {...restProps}>{children}</Break>
}
