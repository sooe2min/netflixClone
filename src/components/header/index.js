import React, { useState } from 'react'
import { Background, Container, Logo, ButtonLink, Group, Link, Profile, Search, SearchIcon, SearchInput, Picture, Dropdown, Feature, FeatureCallOut, Text, PlayButton } from './styles/header'
import { Link as ReactRouterLink } from 'react-router-dom'

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>
}

Header.Search = function HeaderSearch({ searchTurm, setSearchTurm, ...restProps }) {
  const [searchActive, setSearchActive] = useState(false)

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive(!searchActive)}>
        <img src='/images/icons/search.png' alt='search' />
      </SearchIcon>
      <SearchInput
        placeholder='Search films and series'
        value={searchTurm}
        onChange={({ target }) => setSearchTurm(target.value)}
        active={searchActive}
      />
    </Search >
  )
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>
}


Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>
}
