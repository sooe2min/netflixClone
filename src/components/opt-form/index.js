import React from 'react'
import { Container, Text, Input, Button, Break } from './styles/opt-form'

export default function OptForm({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      {children}
    </Container>
  )
}

OptForm.Text = function OptFormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

OptForm.Input = function OptFromInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>
}

OptForm.Button = function OptFormButton({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src='/images/icons/chevron-right.png' alt='Get Started' />
    </Button>
  )
}

OptForm.Break = function OptFormBreak({ children, ...restProps }) {
  return <Break {...restProps} />
}