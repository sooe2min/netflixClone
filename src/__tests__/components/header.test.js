import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Header } from '../../components'
import { MemoryRouter } from 'react-router-dom'
import * as ROUTES from '../../constants/routes'
import logo from '../../logo.svg'

describe('<Header />', () => {
  it('renders the basic <Header /> with a background', () => {
    const { container, getByText, queryByTestId } = render(
      <Header>
        <Header.Frame>
          <Header.Logo to={ROUTES.HOME} src={logo} alt='NetFlix' />
          <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In</Header.ButtonLink>
          <Header.TextLink
            active="true"
            onClick={() => { }}>
            I am a link!
            </Header.TextLink>
        </Header.Frame>
      </Header>
      , { wrapper: MemoryRouter }
    )

    expect(getByText('I am a link!')).toBeTruthy()
    expect(queryByTestId('header-bg')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders the basic <Header /> without a background', () => {
    const { container, getByText, queryByTestId } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo to={ROUTES.HOME} src={logo} alt='NetFlix' />
          <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In</Header.ButtonLink>
          <Header.TextLink
            active="true"
            onClick={() => { }}>
            I am a link!
            </Header.TextLink>
        </Header.Frame>
      </Header>
      , { wrapper: MemoryRouter }
    )

    expect(getByText('I am a link!')).toBeTruthy()
    expect(queryByTestId('header-bg')).toBeFalsy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders the full <Header /> with a background', () => {
    const { container, getByTestId, getByText } = render(
      <Header src="joker1">
        <Header.Frame>
          <Header.Group>
            <Header.Logo to={ROUTES.HOME} src={logo} alt='NetFlix' />
            <Header.TextLink
              active={false}
              onClick={() => { }}>
              Series
            </Header.TextLink>
            <Header.TextLink
              active
              onClick={() => { }}>
              Films
            </Header.TextLink>
          </Header.Group>

          <Header.Group>
            <Header.Profile>
              <Header.Search
                searchTurm="Joker"
                setSearchTurm={() => { }}
              />
              <Header.Picture src="/images/soomin.png" />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src="/images/soomin.png" />
                  <Header.TextLink>Soomin Jang</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink onClick={() => { }}>Sign out</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>

        <Header.Feature>
          <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
          <Header.Text>
            Forever alone in a crowd ..
          </Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>
      </Header>
      , { wrapper: MemoryRouter }
    )

    expect(getByTestId('search-input')).toBeTruthy()
    expect(getByTestId('search-input').value).toBe('Joker')
    fireEvent.change(getByTestId('search-input'), { target: { value: 'Simpsons' } })
    fireEvent.click(getByTestId('search-click'))

    expect(getByText('Series')).toBeTruthy()
    expect(getByText('Films')).toBeTruthy()
    expect(getByText('Soomin Jang')).toBeTruthy()
    expect(getByText('Sign out')).toBeTruthy()
    expect(getByText('Watch Joker Now')).toBeTruthy()
    expect(getByText('Forever alone in a crowd ..')).toBeTruthy()
    expect(getByText('Play')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})