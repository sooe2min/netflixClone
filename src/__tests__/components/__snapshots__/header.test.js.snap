// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> with a background 1`] = `
<div
  class="header__Background-djltcn-0 eZoWCL"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-1 iOsqEC"
  >
    <a
      href="/"
    >
      <img
        alt="NetFlix"
        class="header__Logo-djltcn-3 ebwbHO"
        src="logo.svg"
      />
    </a>
    <a
      class="header__ButtonLink-djltcn-4 itbKHJ"
      href="/signin"
    >
      Sign In
    </a>
    <p
      class="header__Link-djltcn-6 bUcfVy"
    >
      I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the basic <Header /> without a background 1`] = `
<div
  class="header__Container-djltcn-1 iOsqEC"
>
  <a
    href="/"
  >
    <img
      alt="NetFlix"
      class="header__Logo-djltcn-3 ebwbHO"
      src="logo.svg"
    />
  </a>
  <a
    class="header__ButtonLink-djltcn-4 itbKHJ"
    href="/signin"
  >
    Sign In
  </a>
  <p
    class="header__Link-djltcn-6 bUcfVy"
  >
    I am a link!
  </p>
</div>
`;

exports[`<Header /> renders the full <Header /> with a background 1`] = `
<div
  class="header__Background-djltcn-0 fNjhZU"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-djltcn-1 iOsqEC"
  >
    <div
      class="header__Group-djltcn-5 epGjFX"
    >
      <a
        href="/"
      >
        <img
          alt="NetFlix"
          class="header__Logo-djltcn-3 ebwbHO"
          src="logo.svg"
        />
      </a>
      <p
        class="header__Link-djltcn-6 clnqlu"
      >
        Series
      </p>
      <p
        class="header__Link-djltcn-6 clnqlu"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-djltcn-5 epGjFX"
    >
      <div
        class="header__Profile-djltcn-9 kBmErc"
      >
        <div
          class="header__Search-djltcn-10 jFVtuN"
        >
          <button
            class="header__SearchIcon-djltcn-11 lgBHVu"
          >
            <img
              alt="search"
              data-testid="search-click"
              src="/images/icons/search.png"
            />
          </button>
          <input
            class="header__SearchInput-djltcn-12 dREkvG"
            data-testid="search-input"
            placeholder="Search films and series"
            value="Joker"
          />
        </div>
        <button
          class="header__Picture-djltcn-7 hozjoH"
          src="/images/users//images/soomin.png.png"
        />
        <div
          class="header__Dropdown-djltcn-8 dkqRSl"
        >
          <div
            class="header__Group-djltcn-5 epGjFX"
          >
            <button
              class="header__Picture-djltcn-7 hozjoH"
              src="/images/users//images/soomin.png.png"
            />
            <p
              class="header__Link-djltcn-6 clnqlu"
            >
              Soomin Jang
            </p>
          </div>
          <div
            class="header__Group-djltcn-5 epGjFX"
          >
            <p
              class="header__Link-djltcn-6 clnqlu"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-djltcn-1 header__Feature-djltcn-13 iOsqEC huNDmf"
  >
    <h2
      class="header__FeatureCallOut-djltcn-14 CIYaf"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-djltcn-15 cCKdCP"
    >
      Forever alone in a crowd ..
    </p>
    <button
      class="header__PlayButton-djltcn-16 LJejV"
    >
      Play
    </button>
  </div>
</div>
`;
